File: Language/C#/Book/tmp/LINQ实战/LinqInAction/Chapter13.CSharp - LinqBooks/LinqBooks.DataAccess/BookDataAccessObject.cs
#region Imports

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using LinqBooks.Entities;

#endregion Imports

namespace LinqBooks.DataAccess
{
  public class BookDataAccessObject : DataAccessObject
  {
    public List<Book> GetBooksBySubjectName(String subjectName)
    {
      var query =
        from book in DataContext.Books
        where book.SubjectObject.Name == subjectName
        select book;
      return query.ToList();
    }
  }
}

File: Language/C#/Book/LINQ实战/第13章 应用程序各层中的LINQ/Chapter13.VB - LinqBooks/LinqBooks.DataAccess/BookDataAccessObject.vb
Imports LinqBooks.Entities

Public Class BookDataAccessObject
  Inherits DataAccessObject

  Public Function GetBooksBySubjectName(ByVal subjectName As String) As List(Of Book)
    Dim query = _
      From book In DataContext.Books _
      Where book.SubjectObject.Name = subjectName _
      Select book
    Return query.ToList()
  End Function

End Class